src/hooks/spaces/useSpaceBySlug.ts

```diff
@@ -19,48 +24,70 @@ export type UseSpaceBySlugResult = {
  * @param spaceSlug
  * @returns
  */
-export const useSpaceBySlug = (spaceSlug?: string): UseSpaceBySlugResult => {
+export const useSpaceBySlug = (
+  worldSlug?: WorldSlug,
+  spaceSlug?: SpaceSlug
+): UseSpaceBySlugResult => {
   const firestore = useFirestore();
 
   const spacesRef = firestore
     .collection("venues")
-    .where("slug", "==", spaceSlug ?? "")
+    .where("slug", "==", spaceSlug ?? EMPTY_SPACE_SLUG)
     .withConverter(withIdConverter<AnyVenue>());
 
   // Note: Avoid using the option 'initialData' because it will make status always return 'success'
-  const { data: spaces, status } = useFirestoreCollectionData<WithId<AnyVenue>>(
-    spacesRef
-  );
+  const { data: spaces, status: spaceStatus } = useFirestoreCollectionData<
+    WithId<AnyVenue>
+  >(spacesRef);
 
-  const isSpaceLoaded = status !== "loading";
+  const worldsRef = firestore
+    .collection("worlds")
+    .where("slug", "==", worldSlug ?? EMPTY_WORLD_SLUG)
+    .withConverter(withIdConverter<World>());
+  const { data: worlds, status: worldStatus } = useFirestoreCollectionData<
+    WithId<World>
+  >(worldsRef);
```

Comment:
There are functions in `./src/api/space.ts` and `./src/api/world.ts` that might help DRY this